mathics/builtin/forms.py

```diff
@@ -0,0 +1,520 @@
+"""
+Print Forms
+"""
+
+import re
+
+from typing import Optional
+
+
+from mathics.builtin.base import (
+    Builtin,
+    Predefined,
+)
+from mathics.builtin.box.layout import GridBox, RowBox, to_boxes
+from mathics.builtin.comparison import expr_min
+from mathics.builtin.makeboxes import MakeBoxes
+from mathics.builtin.tensors import get_dimensions
+
+from mathics.core.atoms import Integer, String, StringFromPython
+from mathics.core.attributes import (
+    locked as A_LOCKED,
+    protected as A_PROTECTED,
+)
+
+from mathics.core.element import EvalMixin
+from mathics.core.expression import Expression, BoxError
+from mathics.core.list import ListExpression
+from mathics.core.symbols import (
+    Symbol,
+    SymbolFullForm,
+    SymbolList,
+)
+
+from mathics.core.systemsymbols import (
+    SymbolMakeBoxes,
+    SymbolRowBox,
+    SymbolRule,
+)
+
+MULTI_NEWLINE_RE = re.compile(r"\n{2,}")
+
+SymbolNumberForm = Symbol("System`NumberForm")
+SymbolSuperscriptBox = Symbol("System`SuperscriptBox")
+SymbolTableDepth = Symbol("TableDepth")
+
+
+class PrintForms_(Predefined):
+    r"""
+    <dl>
+      <dt>'$PrintForms'
+      <dd>contains the list of basic print forms. It is updated automatically when new 'PrintForms' are defined by setting format values.
+    </dl>
+
+    >> $PrintForms
+     = ...
+
+    Suppose now that we want to add a new format `MyForm`. Initially, it does not belongs to $\$PrintForms$:
+    >> MemberQ[$PrintForms, MyForm]
+     = False
+    Now, let's define a format rule:
+    >> Format[MyForm[F[x_]]]:= "F<<" <> ToString[x] <> ">>"
+    >> Format[F[x_], MyForm]:= MyForm[F[x]]
+    Now, the new format belongs to the $\$PrintForms$ list
+    >> MemberQ[$PrintForms, MyForm]
+     = True
+
+    """
+
+    attributes = A_LOCKED | A_PROTECTED
+    name = "$PrintForms"
+    summary_text = "list the print formats"
+
+    def evaluate(self, evaluation):
+        return ListExpression(*evaluation.definitions.printforms)
```

Comment:
I realize `printforms` was from a prior PR, as I follow this. The setup of printforms in `systemsymbols`  is wrong. 

It doesn't allow the user to create more printforms. 

Instead there should be a class variable indicating whether a Form belowns in `definitions.printforms` and the base class should add that.

Reply:
Actually, if you want to add `MyForm`, you can do something like
```
In[1]:= Format[F[x_],MyForm]:=MyForm[x_]
```
then, `MyForm` is appended to `$PrintForms` (and also should be appended to  `$OutputForms`).

Reply:
Ok  - would you just make the change.